packages/components/src/pages/AnalysisFindings.vue

```diff
@@ -0,0 +1,387 @@
+</template>
+
+<script>
+import VQuickstartLayout from '@/components/quickstart/QuickstartLayout.vue';
+
+export default {
+  name: 'analysis-findings',
+
+  components: {
+    VQuickstartLayout,
+  },
+
+  props: {
+    findings: {
+      default: () => [],
+      type: Array,
+    },
+  },
+
+  data() {
+    return {
+      sortBy: 'name',
+      sortAscending: true,
+      sortedFindingsInfo: this.getUniqueFindings(),
+    };
+  },
+
+  watch: {
+    findings(newVal) {
+      this.findings = newVal;
+    },
+  },
+
+  methods: {
+    sortFindings(sortBy) {
+      if (this.sortBy === sortBy) {
+        this.sortAscending = !this.sortAscending;
+      }
+
+      this.sortBy = sortBy;
+      if (this.sortBy === 'category') {
+        this.sortFindingsByCategory();
+      } else if (this.sortBy === 'name') {
+        this.sortFindingsByName();
+      }
+    },
+
+    sortFindingsByName() {
+      this.sortedFindingsInfo.sort((a, b) => {
+        if (this.sortAscending) {
+          return a[0] < b[0] ? -1 : 1;
+        }
+        return a[0] < b[0] ? 1 : -1;
+      });
+    },
+
+    sortFindingsByCategory() {
+      this.sortedFindingsInfo.sort((a, b) => {
+        if (this.sortAscending) {
+          return a[1] < b[1] ? -1 : 1;
+        }
+        return a[1] < b[1] ? 1 : -1;
+      });
+    },
+
+    getUniqueFindings() {
+      const uniqueFindings = this.groupFindingsByHash();
+
+      return Object.keys(uniqueFindings).map((hash) => {
+        const firstFinding = uniqueFindings[hash][0];
+        const { ruleTitle, impactDomain } = firstFinding.finding;
+
+        return [ruleTitle, impactDomain, hash];
+      });
+    },
+
+    groupFindingsByHash() {
+      return this.findings.reduce((accumulator, finding) => {
+        const hash = finding.finding.hash_v2;
+
+        if (hash in accumulator) {
+          accumulator[hash].push(finding);
+        } else {
+          accumulator[hash] = [finding];
+        }
+
+        return accumulator;
+      }, {});
+    },
+
+    filter(impactDomain) {
+      this.sortedFindingsInfo = this.getUniqueFindings();
+      this.sortFindings(this.sortBy);
+
+      if (impactDomain !== 'all') {
+        this.sortedFindingsInfo = this.sortedFindingsInfo.filter(
+          (info) => info[1] === impactDomain
+        );
+      }
+    },
+
+    openFindingInfo(hash) {
+      this.$root.$emit('open-finding-info', hash);
+    },
+
+    openProblemsTab() {
+      this.$root.$emit('open-problems-tab');
+    },
+  },
+
+  computed: {
+    uniqueFindings() {
+      const seen = {};
+      return this.findings.filter((finding) => {
+        const hash = finding.finding.hash_v2;
+        if (!seen[hash]) {
+          seen[hash] = true;
+          return true;
+        }
+        return false;
+      });
+    },
+
+    findingsByImpactDomain() {
+      return this.uniqueFindings.reduce(
+        (result, finding) => {
+          result[finding.finding.impactDomain].push(finding);
+          return result;
+        },
```

Comment:
What do you mean by "normalize the case of the `impactDomain`"?

Reply:
e.g. `impactDomain.toLowerCase()` would prevent a future rule from breaking this code if it referenced `security` instead of `Security`. [The `ImpactDomain` type in scanner](https://github.com/getappmap/appmap-js/blob/main/packages/scanner/src/types.d.ts#L34) lists the proper casing but I don't feel confident it's actually enforced anywhere when we ingest findings data.

Anyway - not really a necessity, just a thought.

Reply:
Okay, I fixed this and added tests to make sure it handles cases where the `impactDomain` is undefined.